cql3/untyped_result_set.hh

```diff
@@ -188,8 +214,20 @@ public:
         return _rows.back();
     }
 private:
+    using index_map_ptr = ::lw_shared_ptr<untyped_result_set_row::index_map>;
+    using qr_tuple = std::tuple<foreign_ptr<lw_shared_ptr<query::result>>, shared_ptr<const cql3::metadata>>;
+    using storage = std::variant<std::monostate
+        , ::shared_ptr<cql_transport::messages::result_message>
+        , qr_tuple
+    >;
+    struct visitor;
+
+    storage _storage;
+    index_map_ptr _index;
```

Comment:
Does this have to be a pointer? If I understand correctly, both `untyped_result_set::visitor` and `untyped_result_set_row` lifetimes are contained in `untyped_result_set`'s lifetime, so `untyped_result_set` could just store an `index_map` directly and these guys would have a reference to it.

Reply:
I want both types to be movable. Having the index stored inside untyped_result_set wound break refs in the row objects.

Reply:
Then `unique_ptr`?

Reply:
I guess it can be. now that we made the refs in rows actual refs. Not a huge deal though.

Reply:
changed.